### PR TITLE
feat: Use dynamic import for `@react-email/render`

### LINKED ISSUE #243: Remove dependency on react package to make it more lightweight
Hi, first of all thanks for such a great email service, I really like it!

I found that this package is much larger than the packages provided by other email services. Is there any plan to remove the dependency of the `react` package? To make it more lightweight and less stressful for the integration of other frameworks. Or is this package specific to integrating `react`?

I know it's possible to simply use the rest api, but then won't get the benefits of types.

### LINKED ISSUE #309: should not include `react`/`@react-email/render` as a dependency
This is from `pnpm-lock.yaml`:
```
/resend@3.0.0:
	(...)
	dependencies:
	    '@react-email/render': 0.0.11
```

I don't know why `@react-email/render` is a dependency for `resend` package. If I am using resend in a Nuxt project why should i deal with react-email. I think it would be better if `resend` was an independant package. 

_Originally posted by @aboubakrbm in https://github.com/resend/react-email/issues/1209#issuecomment-1902097701_

### LINKED ISSUE #325: Resend dependency broke Vercel deploy
This appears to be of a similar nature to #310. When deploying a Nuxt app using `resend@^3.2.0` I get this error when trying to deploy on Vercel :

Capture d’écran 2024-02-14 à 22 25 58

Everything works fine on local

### LINKED ISSUE #381: v3.3.0 is throwing an error
I saw that `v3.3.0` was made available through npm even though there are no release notes here on GitHub. 

It broke my email sending capability with `next@14.2.4`. 

here's the error which goes back to `@react-email/render@0.0.15` bumped from the prev v0.0.12 

```sh
Error: Dynamic require of "stream" is not supported
    at file:///Users/mfts/dev/mfts/papermark-main/node_modules/@react-email/render/dist/index.mjs:22:9
    at file:///Users/mfts/dev/mfts/papermark-main/node_modules/@react-email/render/dist/index.mjs:104:26
    at Generator.next (<anonymous>)
    at file:///Users/mfts/dev/mfts/papermark-main/node_modules/@react-email/render/dist/index.mjs:41:61
    at new Promise (<anonymous>)
    at __async (file:///Users/mfts/dev/mfts/papermark-main/node_modules/@react-email/render/dist/index.mjs:25:10)
    at readStream (file:///Users/mfts/dev/mfts/papermark-main/node_modules/@react-email/render/dist/index.mjs:94:30)
    at file:///Users/mfts/dev/mfts/papermark-main/node_modules/@react-email/render/dist/index.mjs:133:26
    at Generator.next (<anonymous>)
    at file:///Users/mfts/dev/mfts/papermark-main/node_modules/@react-email/render/dist/index.mjs:41:61
    at new Promise (<anonymous>)
    at __async (file:///Users/mfts/dev/mfts/papermark-main/node_modules/@react-email/render/dist/index.mjs:25:10)
    at onAllReady (file:///Users/mfts/dev/mfts/papermark-main/node_modules/@react-email/render/dist/index.mjs:132:18)
    at finishedTask (/Users/mfts/dev/mfts/papermark-main/node_modules/react-dom/cjs/react-dom-server.node.development.js:6579:5)
    at retryTask (/Users/mfts/dev/mfts/papermark-main/node_modules/react-dom/cjs/react-dom-server.node.development.js:6609:5)
```

### LINKED ISSUE #405: Error: Cannot find module 'react-dom/server'
Hello, I am trying to use resend on my server, but I am getting an error that is preventing me from even starting the server. The error seems to be related to react-email, but I don't understand why this would be necessary. I would like to use HTML for sending emails. Could you please help me understand what I need to do to fix this issue? Thank you for your assistance.

> I also tried downgrading the version from 3.5.0 to 3.4.0 and then to 3.2.0, but the error continued to appear

Error:

```
Error: Cannot find module 'react-dom/server'
Require stack:
- C:\Users\xxx\Desktop\app\server\node_modules\@react-email\render\dist\node\index.js
- C:\Users\xxx\Desktop\app\server\node_modules\resend\dist\index.js
- C:\Users\xxx\Desktop\app\server\dist\utils\resend\resend.module.js
- C:\Users\xxx\Desktop\app\server\dist\app.module.js
- C:\Users\xxx\Desktop\app\server\dist\main.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1144:15)
    at Function.Module._load (node:internal/modules/cjs/loader:985:27)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (C:\Users\xxx\Desktop\app\server\node_modules\@react-email\render\dist\node\index.js:74:30)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Function.Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
```

Deps:
````
    "dependencies": {
        "@nestjs/common": "^10.0.0",
        "@nestjs/config": "^3.2.1",
        "@nestjs/core": "^10.0.0",
        "@nestjs/event-emitter": "^2.0.4",
        "@nestjs/jwt": "^10.2.0",
        "@nestjs/mongoose": "^10.0.4",
        "@nestjs/passport": "^10.0.3",
        "@nestjs/platform-express": "^10.0.0",
        "@nestjs/platform-socket.io": "^10.3.8",
        "@nestjs/throttler": "^5.1.2",
        "@nestjs/websockets": "^10.3.8",
        "bcrypt": "^5.1.1",
        "cookie-parser": "^1.4.6",
        "mongoose": "^8.2.4",
        "nestjs-zod": "^3.0.0",
        "passport": "^0.7.0",
        "passport-jwt": "^4.0.1",
        "reflect-metadata": "^0.2.0",
        "resend": "^3.5.0",
        "rxjs": "^7.8.1",
        "zod": "^3.22.4"
    },
````
    
I just trying to use this dynamic module:

```@Module({})
export class ResendModule {
    public static forRoot(key: string): DynamicModule {
        const resend = new Resend(key);

        const resendProvider: Provider = {
            provide: 'RESEND_CLIENT',
            useValue: resend,
        };

        return {
            module: ResendModule,
            providers: [resendProvider],
            exports: [resendProvider],
            global: true,
        };
    }
}

### PR DESCRIPTION
**Pull request by vcapretz**
use dynamic import for `@react-email/render` only when the `react` prop is used

### CODE REVIEW

File: src/batch/batch.ts [hunk 1]
```diff
@@ -24,9 +23,16 @@ export class Batch {
   ): Promise<CreateBatchResponse> {
     for (const email of payload) {
       if (email.react) {
-        email.html = await renderAsync(email.react as React.ReactElement);
-        // biome-ignore lint/performance/noDelete: <explanation>
-        delete email.react;
+        try {
+          const { renderAsync } = await import('@react-email/render');
```
Comment (gabrielmfern) [inline src/batch/batch.ts]: Would it be worth doing the try-catch around only the `await import`? Might get confusing if the `renderAsync` call itself fails

**Review by gabrielmfern — APPROVED** [review]
👍🏻

**Comment by bukinoshita** [comment]
@vcapretz Can you release an alpha version of it?

**Comment by vcapretz** [comment]
@bukinoshita `resend@4.0.1-alpha.0` is now available to test

**Review by bukinoshita — APPROVED** [review]

**Pull request merged**